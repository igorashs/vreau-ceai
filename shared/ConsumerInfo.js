import styled from 'styled-components';
import AccountSvg from '@/icons/account.svg';
import EmailSvg from '@/icons/email.svg';
import PhoneSvg from '@/icons/phone.svg';
import HomeSvg from '@/icons/home.svg';

const Wrapper = styled.div`
  display: grid;
  gap: 7px;
  padding-top: 7px;
  border-top: 1px solid var(--layout);
`;

const Title = styled.p`
  font-weight: 500;
`;

const Small = styled.small`
  font-weight: 500;
`;

const Info = styled.div`
  display: flex;
  align-items: center;
  gap: 7px;

  svg {
    flex-shrink: 0;
  }
`;

export const ConsumerInfo = ({ name, email, tel, address }) => {
  return (
    <Wrapper>
      <Title>Consumator</Title>
      <Info>
        <AccountSvg />
        <Small>{name}</Small>
      </Info>
      <Info>
        <EmailSvg />
        <Small>{email}</Small>
      </Info>
      <Info>
        <PhoneSvg />
        <Small>{tel}</Small>
      </Info>
      <Info>
        <HomeSvg />
        <Small>{address}</Small>
      </Info>
    </Wrapper>
  );
};
