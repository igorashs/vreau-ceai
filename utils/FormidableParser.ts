import * as validator from '@/utils/validator';
import FormidableNameSpace, { Files, File, IncomingForm } from 'formidable';
import { NextApiRequest } from 'next';
import { Formidable, ProductFields } from 'types';
import { productMessages } from './validator/schemas/product';
import { NextApiSessionRequest } from './withSession';

type ParsedData = {
  fields: Partial<ProductFields>;
  files: Files & { src?: File };
};

type HandleProductForm = (
  req: NextApiRequest | NextApiSessionRequest,
  options: Partial<FormidableNameSpace.Options>,
  allowedFileTypes: string[],
) => Promise<ParsedData>;

const handleProductForm: HandleProductForm = async (
  req,
  options,
  allowedFileTypes,
) => {
  const data: ParsedData = await new Promise((resolve, reject) => {
    const form: Formidable = new IncomingForm(options);

    form.onPart = (part) => {
      if (part.filename) {
        // get file extension
        const fileType = part.mime?.split('/').pop() ?? '';

        // validate allowed file types
        if (!allowedFileTypes.includes(fileType)) {
          form.emit(
            'error',
            validator.createValidationError({
              message: productMessages.src.invalid,
              key: 'src',
            }),
          );
        }
      }

      form.handlePart(part);
    };

    form.parse(req, (err, fields, files) => {
      if (err) {
        // custom error on maxFileSize
        if (err.message.includes('maxFileSize')) {
          reject(
            validator.createValidationError({
              message: productMessages.src.max,
              key: 'src',
            }),
          );
        } else {
          reject(err);
        }
      }

      resolve({ fields, files });
    });
  });

  return data;
};

export default {
  handleProductForm,
};
