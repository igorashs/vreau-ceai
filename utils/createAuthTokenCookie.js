import jwt from 'jsonwebtoken';
import cookie from 'cookie';

const EXPIRE_TIME = 24 * 60 * 60;

export const createAuthTokenCookie = (payload) => {
  const authToken = jwt.sign(payload, process.env.AUTH_PRIVATE_KEY, {
    expiresIn: EXPIRE_TIME
  });

  return cookie.serialize('auth-token', authToken, {
    secure: process.env.NODE_ENV !== 'development',
    httpOnly: true,
    sameSite: 'strict',
    path: '/',
    maxAge: EXPIRE_TIME
  });
};
