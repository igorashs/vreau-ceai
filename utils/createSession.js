import jwt from 'jsonwebtoken';
import cookie from 'cookie';

const EXPIRE_TIME = 24 * 60 * 60;

export const createSession = (payload) => {
  const token = jwt.sign(payload, process.env.AUTH_PRIVATE_KEY, {
    expiresIn: EXPIRE_TIME
  });

  const [header, jwtPayload, signature] = token.split('.');
  const session = header + '.' + jwtPayload;

  const sessionCookie = cookie.serialize('session', session, {
    secure: process.env.NODE_ENV !== 'development',
    sameSite: 'strict',
    path: '/',
    maxAge: EXPIRE_TIME
  });

  const authCookie = cookie.serialize('auth', signature, {
    secure: process.env.NODE_ENV !== 'development',
    httpOnly: true,
    sameSite: 'strict',
    path: '/',
    maxAge: EXPIRE_TIME
  });

  return [sessionCookie, authCookie];
};
