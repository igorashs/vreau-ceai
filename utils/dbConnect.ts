import mongoose from 'mongoose';

const dbConnect = async () => {
  if (mongoose.connection.readyState >= 1) {
    return;
  }

  await mongoose.connect(process.env.MONGODB_URI as string, {
    useNewUrlParser: true,
    useUnifiedTopology: true,
    useCreateIndex: true,
    useFindAndModify: false,
  });
};

export default dbConnect;
