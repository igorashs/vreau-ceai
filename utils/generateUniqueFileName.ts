import { nanoid } from 'nanoid';

const generateUniqueFileName = (fileName: string) => {
  // grab file extension
  const matches = fileName?.match(/\..+$/);

  // create new file name
  const uniqueFileName = nanoid() + (matches && matches[0]);

  return uniqueFileName;
};

export default generateUniqueFileName;
