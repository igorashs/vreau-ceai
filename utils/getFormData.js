export const getFormData = (data) => {
  const formData = new FormData();

  for (const name in data) {
    if (name === 'src') continue;

    formData.append(name, data[name]);
  }

  if (data.src[0]) {
    formData.append('src', data.src[0]);
  }

  return formData;
};
