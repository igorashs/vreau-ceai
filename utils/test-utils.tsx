import { render, RenderOptions } from '@testing-library/react';
import GlobalStyle from 'GlobalStyle';

const Wrapper: React.FC = ({ children }) => (
  <>
    <GlobalStyle />
    {children}
  </>
);

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
) => render(ui, { wrapper: Wrapper, ...options });

export * from '@testing-library/react';

export { customRender as render };
