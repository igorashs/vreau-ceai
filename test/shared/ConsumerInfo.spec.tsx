import { render } from '@/utils/test-utils';
import ConsumerInfo from '@/shared/ConsumerInfo';

describe('ConsumerInfo', () => {
  it('renders correctly', () => {
    const consumer = {
      name: 'qwerty',
      email: '1@1.com',
      tel: '061111111',
      address: 'add1',
    };

    const { container, getByText } = render(<ConsumerInfo {...consumer} />);

    expect(getByText(consumer.name)).toBeVisible();
    expect(getByText(consumer.email)).toBeVisible();
    expect(getByText(consumer.tel)).toBeVisible();
    expect(getByText(consumer.address)).toBeVisible();

    expect(container.firstChild).toMatchSnapshot();
  });
});
