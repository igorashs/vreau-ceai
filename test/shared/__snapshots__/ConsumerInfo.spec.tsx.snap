// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsumerInfo renders correctly 1`] = `
.c0 {
  display: grid;
  gap: 7px;
  padding-top: 7px;
  border-top: 1px solid var(--layout);
}

.c1 {
  font-weight: 500;
}

.c3 {
  font-weight: 500;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 7px;
}

.c2 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

<div
  class="c0"
>
  <p
    class="c1"
  >
    Consumator
  </p>
  <div
    class="c2"
  >
    <svg />
    <small
      class="c3"
    >
      qwerty
    </small>
  </div>
  <div
    class="c2"
  >
    <svg />
    <small
      class="c3"
    >
      1@1.com
    </small>
  </div>
  <div
    class="c2"
  >
    <svg />
    <small
      class="c3"
    >
      061111111
    </small>
  </div>
  <div
    class="c2"
  >
    <svg />
    <small
      class="c3"
    >
      add1
    </small>
  </div>
</div>
`;
